scripts/pythondistdeps.py

```diff
@@ -350,25 +370,29 @@ def normalize_name(name):
                     deps.insert(0, Requirement.parse('setuptools'))
                 # add requires/recommends based on egg/dist metadata
                 for dep in deps:
-                    if normalized_names_require_pep503:
-                        dep_normalized_name = normalize_name(dep.project_name)
-                    else:
-                        dep_normalized_name = dep.key
-
-                    if args.legacy:
-                        name = 'pythonegg({})({})'.format(pyver_major, dep.key)
-                    else:
-                        if args.majorver_only:
-                            name = 'python{}dist({})'.format(pyver_major, dep_normalized_name)
+                    extras_suffixes = [""]
+                    if args.require_extras_subpackages and dep.extras:
+                        extras_suffixes = [f"[{e}]" for e in dep.extras]
```

Comment:
I am a bit lost here. `""` is in the list, but only if `dep.extras` is empty? Should this lien be `+=`?

Reply:
I believe it shouldn't.

* If you have a dependency `foo`, then  `dep.extras` is an empty list and we want to show `python3dist(foo)` (no suffix)
* If you have a dependency `foo[bar]`, then `dep.extras` contains `[bar]` and we want to show `python3dist(foo[bar])` (with the suffix)
* If you have a dependency `foo[bar,baz]`, then `dep.extras` contains `[bar,baz]` and we want to show `python3dist(foo[bar])` and `python3dist(foo[baz])`

Or do you think if we have a dependency on `foo[bar]`, we should require both `foo[bar]` and also plain `foo`?

Reply:
Oh, I was confused about what does this code do.

> Or do you think if we have a dependency on foo[bar], we should require both foo[bar] and also plain foo?

I haven't but now when you mention it, this would certainly be safer (and harmless).

Reply:
Also friendlier for `repoquery --whatrequires` checks.

Reply:
![do it](https://media.giphy.com/media/rjLINlGpJYvvO/200_d.gif)

Reply:
> Also friendlier for `repoquery --whatrequires` checks.

Right, that's a good point. On it.

Reply:
Done